gammapy/image/core.py

```diff
@@ -444,27 +447,124 @@ def pad(self, mode, pad_to_factor=None, pad_width=None, **kwargs):
         >>> image2.data.shape
         (16, 12)
         """
-        if pad_to_factor is not None and pad_width is not None:
-            raise ValueError('Indicate only one parameter: '
-                             'either "pad_width" or "pad_to_factor"')
-        if pad_to_factor is None and pad_width is None:
+        # converting from unicode to ascii string as a workaround
+        # for https://github.com/numpy/numpy/issues/7112
+        mode = str(mode)
+        wcs = self.wcs.deepcopy()
+
+        args_none = np.array([_ is None for _ in [pad_width, pad_to_factor, shape]])
+
+        if args_none.all():
             raise ValueError('One parameter must be indicated: '
-                             'either "pad_width" or "pad_to_factor"')
+                             'either "pad_width", "pad_to_factor" or "shape"')
+        if args_none.sum() < 2:
+            raise ValueError('Indicate only one parameter: '
+                             'either "pad_width", "pad_to_factor" or "shape"')
 
         if pad_to_factor is not None:
             pad_width = pad_to_factor - (np.array(self.data.shape) % pad_to_factor)
             pad_width = [(0, pad_width[0]), (0, pad_width[1])]
 
-        # converting from unicode to ascii string as a workaround
-        # for https://github.com/numpy/numpy/issues/7112
-        mode = str(mode)
+        if shape is not None:
+            xdiff = shape[1] - self.data.shape[1]
```

Comment:
This line is obsolete, but I will use it in other places.

I think right now I would make it private, but later one could think about adding some further info to the `Axis` object (e.g. `SkyImage.axes.x.unit`, `SkyImage.axes.x.idx` or `SkyCube.axes.energy.unit`) and make it public.